docs/certificate_management.md

```diff
@@ -0,0 +1,228 @@
+	ObjectMeta: metav1.ObjectMeta{
+		GenerateName: "osm-",
+		Namespace:    cm.namespace,
+	},
+	Spec: cmapi.CertificateRequestSpec{
+		Duration: duration,
+		IsCA:     false,
+		Usages: []cmapi.KeyUsage{
+			cmapi.UsageKeyEncipherment, cmapi.UsageDigitalSignature,
+		},
+		Request:   csrPEM,
+		IssuerRef: cm.issuerRef,
+	},
+}
+```
+
+The certificate will be retrieved by making a request directly to cert-manager by `(*CertManager).certificaterFromCertificateRequest`. Here is an example issuer for OSM:
+
+```
+$ kubectl get issuer -n osm-system osm-ca -o yaml
+metadata:
+  annotations:
+    kubectl.kubernetes.io/last-applied-configuration: |
+      {"apiVersion":"cert-manager.io/v1","kind":"Issuer","metadata":{"annotations":{},"name":"osm-ca","namespace":"osm-system"},"spec":{"ca":{"secretName":"osm-ca-bundle"}}}
+  name: osm-ca
+  namespace: osm-system
+spec:
+  ca:
+    secretName: osm-ca-bundle
+```
+
+In this case the CA root cert is defined as having its own issuer which is a self-signed certificate:
+
+```
+$ kubectl get cert -n osm-system osm-ca -o yaml
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  annotations:
+    kubectl.kubernetes.io/last-applied-configuration: |
+      {"apiVersion":"cert-manager.io/v1","kind":"Certificate","metadata":{"annotations":{},"name":"osm-ca","namespace":"osm-system"},"spec":{"commonName":"osm-system","duration":"2160h","isCA":true,"issuerRef":{"group":"cert-manager.io","kind":"Issuer","name":"selfsigned"},"secretName":"osm-ca-bundle"}}
+  name: osm-ca
+  namespace: osm-system
+spec:
+  commonName: osm-system
+  duration: 2160h0m0s
+  isCA: true
+  issuerRef:
+    group: cert-manager.io
+    kind: Issuer
+    name: selfsigned
+  secretName: osm-ca-bundle
+```
+
+The certificate is set as a CA cert (you can see this in the certificate metadata) and the backing secret name is set to `osm-ca-bundle`. You can inspect the CA bundle:
+
+```
+$ kubectl get secret -n osm-system osm-ca-bundle -o yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  annotations:
+    cert-manager.io/alt-names: ""
+    cert-manager.io/certificate-name: osm-ca
+    cert-manager.io/common-name: osm-system
+    cert-manager.io/ip-sans: ""
+    cert-manager.io/issuer-group: cert-manager.io
+    cert-manager.io/issuer-kind: Issuer
+    cert-manager.io/issuer-name: selfsigned
+    cert-manager.io/uri-sans: ""
+    manager: controller
+    operation: Update
+  name: osm-ca-bundle
+  namespace: osm-system
+data:
+  ca.crt: LS0tLS1CRUdJTiBDRVJUSUZJQ...
+  tls.crt: LS0tLS1CRUdJTiBDRVJUSUZJ...
+  tls.key: LS0tLS1CRUdJTiBSU0EgUFJJ...
+```
+
+## HashiCorp Vault
+
+When utilizing Vault as the certificate manager, OSM will make a call to Vault's API through `Logical.Write` to issue certificates. The returned certificate will be sent to the caller. To get the issuing CA from Vault, OSM will create a temporary certificate with `localhost` as the CommonName (CN). This temporary certificate will be immediately released.
```

Comment:
Can we document the options available to configure Vault? For example, 

- **osm.certificateProvider.kind** - Certificate manager type (for vault, use `vault`)
- **osm.vault.host** - Hashicorp Vault host/service - where Vault is installed (defaults to "")
- **osm.vault.protocol** - The protocol to use to connect to Vault (defaults to "http")
- **osm.vault.token** - The token that should be used to connect to Vault (defaults to "")
- **osm.vault.role** -   The vault role to be used by Open Service Mesh (defaults to "openservicemesh")